### ISSUE TITLE
FormRequest.formdata with GET method duplicates same key in query string

### ISSUE THREAD

**Issue by talhashraf** [scrapy/scrapy#2919]
```
>>> from scrapy import FormRequest
>>> FormRequest('http://example.com/?id=9', method='GET', formdata={'id': '8'}).url
'http://example.com/?id=9&id=8'
```
Notice how `id` is being duplicated in query string. It must replace the value if same key is found. I believe this can be handled in [form.py#L36](https://github.com/scrapy/scrapy/blob/master/scrapy/http/request/form.py#L36).

**Comment by kmike** [comment]
Yes, this looks like a bug.

**Comment by jslay88** [comment]
Hopefully that works for you. First time making a PR, so please excuse any errors.

**Comment by keremgocen** [comment]
anyone working on this?

**Comment by Gallaecio** [comment]
I’m not sure this should be considered a bug. Specially after giving some though to https://github.com/scrapy/scrapy/pull/4730.

**Comment by Gallaecio** [comment]
Fixed by #3579
